build.gradle

```diff
@@ -204,117 +216,83 @@ repositories {
+
+  testImplementation group: 'io.rest-assured', name: 'rest-assured', version: versions.restAssured
+  testImplementation group: 'io.rest-assured', name: 'xml-path', version: versions.restAssured
+  testImplementation group: 'io.rest-assured', name: 'json-path', version: versions.restAssured
 
-  //implementation group: 'com.github.hmcts',  name: 'auth-checker-lib', version: '2.1.5'
 
-  testImplementation(platform('org.junit:junit-bom:5.8.2'))
-  testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
   testImplementation group: 'org.springframework.boot', name: 'spring-boot-starter-test', {
     exclude group: 'junit', module: 'junit'
     exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
   }
-  testImplementation "org.junit.vintage:junit-vintage-engine:${versions.junit}"
 
-  testImplementation group: 'org.pitest', name: 'pitest', version: versions.pitest
-  testImplementation 'info.solidsoft.gradle.pitest:gradle-pitest-plugin:1.5.2'
-  testImplementation 'org.codehaus.sonar-plugins:sonar-pitest-plugin:0.5'
+  testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter', version: versions.junitJupiter
 
-  functionalTestImplementation sourceSets.main.runtimeClasspath
-  functionalTestImplementation sourceSets.test.runtimeClasspath
+  testImplementation group: 'org.pitest', name: 'pitest', version: versions.pitest
+  testImplementation group: 'info.solidsoft.gradle.pitest', name: 'gradle-pitest-plugin',
+          version: versions.gradlePiTestPlugin
+  testImplementation group: 'org.codehaus.sonar-plugins', name: 'sonar-pitest-plugin',
+          version: versions.sonarPiTestPlugin
 
-  smokeTestImplementation sourceSets.main.runtimeClasspath
-  smokeTestImplementation sourceSets.test.runtimeClasspath
+  testImplementation group: 'org.assertj', name: 'assertj-core', version: versions.assertJ
+}
 
-  constraints {
-    implementation(group: 'com.jhlabs', name: 'filters', version: '2.0.235-1') {
-      because 'CVE-2005-0406'
-    }
-    implementation(group: 'com.ibm.icu', name: 'icu4j', version: '70.1') {
-      because 'CVE-2020-21913'
-    }
-    implementation(group: 'org.apache.pdfbox', name: 'pdfbox', version: '2.0.25') {
-      because 'CVE-2018-11797,CVE-2018-8036,CVE-2021-27807,CVE-2021-27906,CVE-2021-31811,CVE-2021-31812'
-    }
-    implementation(group: 'org.apache.pdfbox', name: 'xmpbox', version: '2.0.25') {
-      because 'CVE-2018-11797, CVE-2018-8036, CVE-2021-27807, CVE-2021-27906, CVE-2021-31811, CVE-2021-31812'
-    }
-    implementation(group: 'org.bouncycastle', name: 'bcprov-jdk15on', version: '1.64') {
-      because 'CVE-2020-15522'
-    }
-    implementation(group: 'com.thoughtworks.xstream', name: 'xstream', version: ' 1.4.19') {
-      because 'CVE-2021-43859'
-    }
+dependencyManagement {
+  imports {
+    mavenBom "org.springframework.cloud:spring-cloud-dependencies:${versions.springCloud}"
```

Comment:
What is the benefit/difference between using dependencyManagement and constraints?

Reply:
Oops, I need to check my Github email rule.

I'm only using Spring's dependency management here to control what versions of Spring Cloud get resolved. If you noticed I haven't provided an explicit 3.1.0 version for spring-cloud-starter. The BOM controls what versions get put in.

I was going to use constraints to fix transitive dependency versions but John suggested that we should suppress them instead to reduce the chance of any incompatibilities arising.